ui/job-view/PushJobs.jsx

```diff
@@ -1,24 +1,55 @@
-import _ from 'lodash';
 import React from 'react';
 import PropTypes from 'prop-types';
 
 import { thPlatformMap, thSimplePlatforms, thEvents } from '../js/constants';
-import { getPushTableId, getPlatformRowId } from '../helpers/aggregateId';
-import Platform from './Platform';
+import { getPlatformRowId, getPushTableId } from '../helpers/aggregateId';
 import { findInstance, findSelectedInstance, findJobInstance } from '../helpers/job';
 import { getUrlParam } from '../helpers/location';
 import { getLogViewerUrl } from '../helpers/url';
 import JobModel from '../models/job';
+import Platform from './Platform';
 
 export default class PushJobs extends React.Component {
+  static getDerivedStateFromProps(nextProps, state) {
+    const { filterModel, push } = nextProps;
+    const { platforms } = state;
+    const selectedJobId = parseInt(getUrlParam('selectedJob'));
+
+    if (!platforms.length) return null;
+
+    const newPlatforms = platforms.reduce((acc, platform) => (
+      [...acc, PushJobs.filterPlatform(platform, selectedJobId, push, filterModel)]
+    ), []);
+
+    return { platforms: newPlatforms };
+  }
+
+  static filterPlatform(platform, selectedJobId, push, filterModel) {
```

Comment:
This function didn't really change.  It's just now static so I can use it in ``getDerivedStateFromProps``.  But it had to be moved before the constructor due to the ordering rule.